mmf/utils/build.py

```diff
@@ -226,7 +226,16 @@ def build_optimizer(model, config):
             )
 
     parameters = get_optimizer_parameters(model, config)
-    optimizer = optimizer_class(parameters, **params)
+
+    if optimizer_config.get("enable_state_sharding", False):
+        from fairscale.optim.oss import OSS
```

Comment:
We can also do a `try`, `except ImportError` and ask user to install the fairscale library separately if we don't want explicit dependency on it. This would also solve some things on fbcode side, user who want it should include fairscale in the targets. Adding more dependency always causes more pain.